### PR TITLE
opamp-usage blog

### PR DESCRIPTION
**Pull request by nemoshlag**

### CODE REVIEW

**Comment by svrnm** [comment]
@nemoshlag , I fixed the header for you

here's the preview: https://deploy-preview-1847--opentelemetry.netlify.app/blog/2022/opamp/

**Comment by nemoshlag** [comment]
Thanks @svrnm!

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:19]: you don't need that on the otel blog.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:39]: Remove the `####`

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:105]: Remove "Sorry, no error handling here."

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:173]: shown. <newline><newline> It's time ...

Create a new paragraph

**Review by svrnm — CHANGES_REQUESTED** [review]

**Comment by svrnm** [comment]
Some initial comments, will do some more tomorrow.

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [reply content/en/blog/2022/opamp/index.md:19]: Note: line 19 - OpenTelemetry is misspelled.

@cartermp, @svrnm, @austinlparker and @chalin: I was wondering if we've standardized on how we spell/use 'open source'? Is it "open source", "open-source", "Open Source" etc.   
(this has come up in another, ie work related, context and was the subject of much debate!)

Comment (melbgirl) [reply content/en/blog/2022/opamp/index.md:19]: I believe "OTel" (rather than "Otel") is correct.

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:13]: For consistency, please replace "our service" with "a service" (our you could instead replace "a client" with "our client", but in this case an impersonal subject is more appropriate to the sentence as it's discussing the situation in general).

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:181]: "verity it**s** running"

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [reply content/en/blog/2022/opamp/index.md:173]: Also suggest marking up the URL for readability, eg 'http://localhost:8080/'

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:186]: "edit **the** 'supervisor.yaml' agent executable to point ..."

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:185]: Please reformat this to show the commands in the terminal, rather than expressing as text. ie

Next, open a new terminal and run the following commands
```console
foo@bar: cd internal/examples/supervisor/bin 
foo@bar: edit supervisor.yaml 
```
etc

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:190]: Remove "Great"

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:201]:
```console

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:228]: New paragraph after the "itself."
Then, start next sentence should be "What do we have here?"

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:232]: "What's next?"

File: content/en/blog/2022/opamp/index.md
Comment (melbgirl) [inline content/en/blog/2022/opamp/index.md:239]: Suggest rewriting the paragraph for readability:
"This is a very basic implementation. Wrapping a system above OpAMP achieves some of the instrumentation orchestrator vision. You're starting with being able to externally add and match tailor-made dynamic telemetry for your system. Imagine what AI can achieve on this type of system? It could collect metrics over the entire system, automatically and dynamically adding trace/log collections onto any detected bottlenecks. 
Using this protocol enables many new possibilities: I believe it has the potential to change how we think about telemetry."

**Review by melbgirl — CHANGES_REQUESTED** [review]
Changes around readabilty of the contents, including using the console markdown for terminal commands.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:14]: >  In an ideal world

I have my issues with that statement, ideal compared to what? I think it would be helpful to go into a little bit more detail about "we would modify the entire service’s telemetry without the need for code changes or deployment."

So maybe start a new paragraph and outline the problem that right now with OpenTelemetry you can not do that, and what the implications are (=> service restart, no capability to change behavior on runtime, etc.)

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:15]: > ... I believe the time has arrived.

I would also think that rephrasing that is necessary, that makes it sound like an announcement for OpAMP being stable, which is not the case.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md:19]: > Note: line 19 - OpenTelemetry is misspelled.
> 
> @cartermp, @svrnm, @austinlparker and @chalin: I was wondering if we've standardized on how we spell/use 'open source'? Is it "open source", "open-source", "Open Source" etc. (this has come up in another, ie work related, context and was the subject of much debate!)

is there something from the LF/CNCF on that? We stick with that.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:28]: Link to the OpAMP spec here.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md:185]: This assumes that people edit files on the console, of course they should be able to use whatever editor they like, so go with "Edit `supervisor.yaml` in `internal/examples/supervisor/bin`. 

Give an inline yml that shows how it should look like.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:186]: Now build and run the supervisor in `internal/examples/supervisor`:

```
$ go build -o bin/supervisor main.go
$ cd bin
$ ./supervisor
```

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:194]: We can now see our agent

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:216]: port 8080? is there a space missing?

**Review by svrnm — CHANGES_REQUESTED** [review]

**Comment by svrnm** [comment]
@open-telemetry/opamp-go-approvers, @open-telemetry/opamp-spec-approvers

File: content/en/blog/2022/opamp/index.md
Comment (nemoshlag) [reply content/en/blog/2022/opamp/index.md:239]: Thanks @melbgirl! It looks so much better now

File: content/en/blog/2022/opamp/index.md
Comment (tigrannajaryan) [inline content/en/blog/2022/opamp/index.md:21]:
```suggestion
telemetry without those barriers? Thanks to
```

File: content/en/blog/2022/opamp/index.md
Comment (tigrannajaryan) [inline content/en/blog/2022/opamp/index.md:28]:
```suggestion
[implementation](https://github.com/open-telemetry/opamp-go/) is at the Beta stage. It allows configuration changes as well as
```

File: content/en/blog/2022/opamp/index.md
Comment (tigrannajaryan) [inline content/en/blog/2022/opamp/index.md:61]:
```suggestion
Next, add a basic file named “effective.yaml” configuration file. Place it in the
```

File: content/en/blog/2022/opamp/index.md
Comment (tigrannajaryan) [inline content/en/blog/2022/opamp/index.md:192]:
```suggestion
Next, edit supervisor.yaml to point at our agent. It should look like this:
```

**Review by tigrannajaryan — COMMENTED** [review]
Thanks for the blog post @nemoshlag !

I left a few minor suggestions.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: resolved

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md:214]:
```suggestion
```console
```

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: resolved

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: fixed via the [other suggestion below ](https://github.com/open-telemetry/opentelemetry.io/pull/1847#discussion_r992842915)

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: suggestion accepted and in the PR. resolved

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: suggestion was added by @nemoshlag

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: suggestion was added by @nemoshlag

**Review by svrnm — APPROVED** [review]

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: @nemoshlag this is still relevant, please review.

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md]:
```suggestion
We can now see our agent running at http://localhost:4321/. Select it and pass
```

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:171]:
```suggestion
Build it with `go build .` Now if we run it and open [http://localhost:8080/](http://localhost:8080/) on
```

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: replaced by https://github.com/open-telemetry/opentelemetry.io/pull/1847/files#r994285865

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: outdated

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [inline content/en/blog/2022/opamp/index.md:241]:
```suggestion
These prints are the trace itself! 

To sum up, we have here a server that
```

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: copied @melbgirl's comment over into a new suggestion: https://github.com/open-telemetry/opentelemetry.io/pull/1847/files#r994288154

File: content/en/blog/2022/opamp/index.md
Comment (svrnm) [reply content/en/blog/2022/opamp/index.md]: suggestion applied by @nemoshlag

**Comment by svrnm** [comment]
All, apologies for the mess with opening, closing, re-opening suggestions, I was led astray by a view of an old commit.

@nemoshlag I updated all the remaining suggestions, so you can review them and add them if you think they make sense.

@melbgirl , @tigrannajaryan thanks for your reviews!

**Review by chalin — CHANGES_REQUESTED** [review]
There's a file named `content/en/blog/2022/opamp/img.png` in this PR but it doesn't appear to be used in the blog post and GH isn't showing a preview of what it is. Maybe this was committed by mistake?

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:40]:
```suggestion
```go
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:61]:
```suggestion
Next, add a basic configuration file named `effective.yaml`. Place it in
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:180]:
```suggestion
Visit <http://localhost:4321> to verify server is running. Notice that no agent
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:173]:
```suggestion
Build and run this app:

```shell
go build .
<insert run command here>
\``` <-- remove the slash here after merge this suggestion

Open a browser and visit <http://localhost:8080>. Nothing special will be shown.

It’s time to add some opamp. Git clone
[opamp-go](https://github.com/open-telemetry/opamp-go) and run the server with:
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:197]:
```suggestion
```shell
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:207]:
```suggestion
Via the supervisor we can now see our agent running at <http://localhost:4321>.
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:226]:
```suggestion
Finally, visit <http://localhost:8080>. Traces should now appear in
`internal/examples/supervisor/bin/agent.log`.
```

File: content/en/blog/2022/opamp/index.md
Comment (chalin) [inline content/en/blog/2022/opamp/index.md:241]: I'm unsure what you mean here, maybe:

```suggestion
These lines are the trace itself! 
```

File: content/en/blog/2022/opamp/index.md
Comment (nemoshlag) [reply content/en/blog/2022/opamp/index.md]: Thanks @chalin, I've merged this change and removed the accidentally committed img file

**Review by chalin — APPROVED** [review]
LGTM for the updates done. Haven't done a full read, but I believe that @svrnm et al. have.

**Comment by svrnm** [comment]
I would like to get @tigrannajaryan's (or any other opamp maintainer/approver) to give us an approval before we merge.

we also need a @open-telemetry/blog-approvers (@austinlparker!)

**Review by tigrannajaryan — APPROVED** [review]
Thanks @nemoshlag for writing this and everyone for reviewing it.

**Review by austinlparker — APPROVED** [review]
lgtm

**Comment by svrnm** [comment]
Awesome. Then we will have the gcp blog on Monday and this one on Wednesday. 

@nemoshlag please update the date accordingly

**Pull request merged**
